### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by xsawyerx** [xsawyerx/pakket#214]
I would like to collect the goals of the documentation here.

- [ ] All classes
- [ ] Introduction
- [ ] Tutorial

Anything else?

**Comment by reyjrar** [comment]
Going to ping this item.  I'm looking for a solution to do all of this, but the missing documentation is going to be a problem.  Happy to help write it, but I don't even know how to get started.  Been bumbling around in the POD a bit and I think I get adding perl modules, but it's still unclear if Pakket helps manage Perl itself, or potentially N+1 Perl's.

**Comment by xsawyerx** [comment]
Sorry for the late reply. (This isn't my primary project at the moment.)

While we could sit and write all of the documentation, it will take time. Instead, I think we should focus on what your needs are and progress based on that.

There is a talk I gave about the project at TPC: https://www.youtube.com/watch?v=-WBL0QoQ3kw

Could you please detail your use-case in mind for Pakket and I could help guide through the project on whether what you want to do is possible and how to best approach it. Throughout this, we could also add the missing bits of documentation to address your needs at any given time.

Does that sound like a plan to you?

**Comment by reyjrar** [comment]
Yeah, I'll check out the video of the talk. My use-case is an off-system Perl + module distribution for my production systems.  We're distributing everything via RPM and I want to move away from that since we're duplicating all the package metadata CPAN already provides.  RPM's are slow to deploy and the specfiles suck so it makes the whole process incredibly opaque to developers.

Ideally, I'd like to use Pakket to distribute N versions of Perl and role-dependent sets of modules to servers across my infrastructure. I wasn't sure how to start with it, my best guess would be something like:

1. Build Pakket with the System Perl
2. Use Pakket with System Perl to Build Perl 5.28.1
3. Use Pakket with System Perl to populate modules to 5.28.1
4. Use Pakket with System Perl to distribute Perl 5.28.1 + modules to systems
5. Applications use the Perl 5.28.1 to run happy applications

**Comment by xsawyerx** [comment]
It sounds like Pakket would serve your purpose.

Pakket has several forms to use it, but the two recommended ones do not require building it:

1. You can run the Seacan script. This creates a self-contained instance of Pakket, including its own Perl interpreter (not the one you would use to build stuff, not to worry), all the Pakket dependencies (so you don't have to package and distribute them yourself), and the Pakket code. At Booking, we create an RPM of that as the Pakket system. You can package it however you want, or even run this directly on the build machine.

2. If you do not have the minimum modules to run Seacan, this script and all of its dependencies have also been grouped together in a fatpacked file. This requires zero set-up to build Pakket.

Let's try to find time to talk about this over video as well. That would likely be faster and could help us surface what we want to document and how.
